Source/SvgExtentions.cs

```diff
@@ -50,10 +50,15 @@ public static string GetXML(this SvgElement elem)
             try
             {
                 Thread.CurrentThread.CurrentCulture = CultureInfo.InvariantCulture;
+
+                var writerSettings = new XmlWriterSettings { Encoding = System.Text.Encoding.UTF8 };
+
                 using (var str = new StringWriter())
-                using (var xml = new XmlTextWriter(str))
                 {
-                    elem.Write(xml);
+                    using (var xml = XmlWriter.Create(str, writerSettings))

```

Comment:
There seems to be no need to separate code block.

Reply:
Need to either close the XmlWriter or explicitly call XmlWriter.Flush() before reading str.
Closing the using block is a convenient way to force the writer to flush.

Previous code did not flush XmlTextWriter but seemed to get away with it.

Reply:
Both look equivalent in code.
I think it's less misleading to call the `Flush` method explicitly.

Similarly, in order to complete write process, please call the Flush method in the following location as well.
https://github.com/svg-net/SVG/blob/ef55e3c2e5436bc37ca328416c2003ce71b020f6/Source/SvgElement.cs#L965

Reply:
Changed SvgExtentions.cs to explicitly call Flush().

I don't see any need or benefit in changing SvgElement.cs... it will be inefficient to flush each element individually.

Reply:
> I don't see any need or benefit in changing SvgElement.cs...

In the following case.
```cs
var element = new SvgDescription();
var str = new StringWriter();
var writer = new XmlTextWriter(str);
element.Write(writer);
```

In this case, `XmlTextWriter` can get string immediately after, but `XmlWriter` cannot get string until `Flush` method is called.

> it will be inefficient to flush each element individually.

I think it's inefficient when called from `SvgDocument`.
It seems that need to decide whether to flush.

Reply:
It should not be the responsibility of `SvgElement` to flush the writer.

It is the calling client's responsibility to ensure the writer is flushed. This happens when the writer is closed, so is not normally a problem. If a different policy is required for some specific use case, then a custom XmlWriter class can be created to implement the desired policy.

Reply:
> It should not be the responsibility of SvgElement to flush the writer.

Please make it clear.
If there is a guideline etc. in MS, it is better to put the link in xmldoc.

At the very least, this PR will make the above code incompatible.

Reply:
I don't understand what you are asking for.

I have changed the API of this library to accept an `XmlWriter` where previously it would only accept an `XmlTextWriter`.
However, if you pass in an `XmlTextWriter`, it will behave exactly as before. The only change is that you _may_ now pass another kind of `XmlWriter`.

So:
```csharp
var element = new SvgDescription();
var str = new StringWriter();
var writer = new XmlTextWriter(str);
element.Write(writer);
```
works as before.

The behaviour will only change if you choose to do:
```csharp
var element = new SvgDescription();
var str = new StringWriter();
var writer = XmlWriter.Create(str);    // factory method for abstract XmlWriter
element.Write(writer);
```
in which case you are passing in an abstract `XmlWriter` instead.

Reply:
> I have changed the API of this library to accept an `XmlWriter` where previously it would only accept an `XmlTextWriter`.

This PR is so.
But .Net recommends using `XmlWriter`.

https://docs.microsoft.com/dotnet/api/system.xml.xmltextreader
> Starting with the .NET Framework 2.0, we recommend that you use the `XmlWriter` class instead.

Until now, you could only use `XmlTextWriter`.
But This PR will allow to use `XmlWriter`.

Users can take this opportunity to use `XmlWriter`.

It would be nice if `XmlWriter` could get the same result as `XmlTextWriter`, but that would be [inefficient](https://github.com/svg-net/SVG/pull/824#discussion_r570074069), so I thought it would be nice to specify in xmldoc(remarks ??) who would `Flush`.

Reply:
Added XmlDoc commentary with links to MS docs.